server/app/models/analysis.rb

```diff
@@ -306,6 +318,22 @@ def status
     end
   end
 
+  # update the job status to indicate that postprocessing is complete.
+  # used from finalize method which is only called for environments using resque
+  def complete_postprocessing!
+    j = jobs_status
+    begin
+      raise "Postprocessing should only happen in environments that use Resque for job management." unless Rails.application.config.job_manager == :resque
+      job = j.last
+      raise "Attempt to complete postprocessing for job with status '#{job[:status]}'.  Only permitted for status 'completed'." unless job[:status] == 'completed'
+      job[:status] = 'post-processing completed'
```

Comment:
I'm testing to make that this doesn't cause any issues.

Reply:
so per travis... there is a test failure due to the new state. I'm concerned that this new "state" will cause PAT or other third party applications to fail.

![image](https://user-images.githubusercontent.com/1907354/49881472-b6c05500-fdeb-11e8-9710-ae11a2786772.png)

Reply:
As far as PAT goes, I think this would be a problem (I think we cue off of the 'completed' word).  But PAT would be ok if the status was another word that didn't include 'completed'.

Reply:
so the 'postprocessing-completed' status doesn't go through to PAT, as it gets converted to 'completed' in the [Analysis#status method](https://github.com/NREL/OpenStudio-server/blob/analysis-complete/server/app/models/analysis.rb#L308).  So as is, the 'post-processing completed' is only internal.  the 'post-processing' is external.

If it's critical not to add a new status, i could return an existing status 'started' during post-processing and set a flag when post-processing completes so i can return 'completed' correctly.

however, that wouldn't fix this test, which seems to indicate either 
1) analysis finalization isn't running for this test environment, although it is using resque, or
2) the code to modify status after finalization runs isn't working.

i'm trying now to better understand the env for this docker test

Reply:
@nllong let's chat about docker testing. 
i've confirmed that the environment docker-test assumes resque for job management, but i don't think we actually use the queuing.  at least, i don't see workers running or even a web-background container.  i'll try to use this issue as a usecase to get the integration tests for docker running...
 relevant files:
1) [travis.yml with test commands](https://github.com/NREL/OpenStudio-server/blob/develop/.travis.yml#L57)
2) [docker.test.yml](https://github.com/NREL/OpenStudio-server/blob/develop/docker-compose.test.yml) i don't understand how we can have a redis queue without resque workers  and a web-background container
3) i'm guessing it's magic in our rspec files... we probably aren't using resque at all.  if we want to leave that way, we can change the docker-test environment to use a different job manager and bypass the 'post-processing' status
4) but this is the place to actually try running something with the full stack, including resque workers.

Reply:
ok, we actually set the job_manager explicitly to resque in [this spec](https://github.com/NREL/OpenStudio-server/blob/develop/server/spec/features/resque_run_simulation_data_point_spec.rb#L41)